rust/rsc/src/common/database.rs

```diff
@@ -0,0 +1,131 @@
+        key: Set(key),
+        desc: Set(description),
+    };
+
+    active_key.insert(db).await
+}
+
+// ----------             Read             ----------
+pub async fn read_api_key(
+    db: &DatabaseConnection,
+    id: Uuid,
+) -> Result<Option<api_key::Model>, DbErr> {
+    api_key::Entity::find_by_id(id).one(db).await
+}
+
+pub async fn read_api_keys(db: &DatabaseConnection) -> Result<Vec<api_key::Model>, DbErr> {
+    api_key::Entity::find().all(db).await
+}
+
+// ----------            Update            ----------
+
+// ----------            Delete            ----------
+pub async fn delete_api_key(
+    db: &DatabaseConnection,
+    key: api_key::Model,
+) -> Result<DeleteResult, DbErr> {
+    tracing::info!("Deleting api key {}", key.key);
+    api_key::Entity::delete_by_id(key.id).exec(db).await
+}
+
+// --------------------------------------------------
+// ----------       Local Blob Store       ----------
+// --------------------------------------------------
+
+// ----------            Create            ----------
+pub async fn create_local_blob_store(
+    db: &DatabaseConnection,
+    root: String,
+) -> Result<local_blob_store::Model, DbErr> {
+    let active_blob_store = blob_store::ActiveModel {
+        id: NotSet,
+        r#type: Set("LocalBlobStore".to_string()),
+    };
+
+    let store = active_blob_store.insert(db).await?;
+
+    let active_local_blob_store = local_blob_store::ActiveModel {
+        id: Set(store.id),
+        created_at: NotSet,
+        root: Set(root),
+    };
+
+    active_local_blob_store.insert(db).await
+}
+
+// ----------             Read             ----------
+pub async fn read_local_blob_store(
+    db: &DatabaseConnection,
+    id: Uuid,
+) -> Result<Option<local_blob_store::Model>, DbErr> {
+    local_blob_store::Entity::find_by_id(id).one(db).await
+}
+
+pub async fn read_local_blob_stores(
+    db: &DatabaseConnection,
+) -> Result<Vec<local_blob_store::Model>, DbErr> {
+    local_blob_store::Entity::find().all(db).await
+}
+
+// ----------            Update            ----------
+
+// ----------            Delete            ----------
+pub async fn delete_local_blob_store(
+    db: &DatabaseConnection,
+    store: local_blob_store::Model,
+) -> Result<DeleteResult, DbErr> {
+    tracing::info!("Deleting local blob store {}", store.id);
+    local_blob_store::Entity::delete_by_id(store.id)
+        .exec(db)
+        .await?;
+    blob_store::Entity::delete_by_id(store.id).exec(db).await
+}
+
+// --------------------------------------------------
+// ----------             Job              ----------
+// --------------------------------------------------
+
+// ----------            Create            ----------
+
+// ----------             Read             ----------
+
+// ----------            Update            ----------
+
+// ----------            Delete            ----------
+pub async fn delete_all_jobs(db: &DatabaseConnection) -> Result<DeleteResult, DbErr> {
```

Comment:
It's not a safe action to arbitrarily delete all the blobs for two reasons.
1) The deletion process needs to make the filesystem deletes/api calls to whatever remote bucket.
2) You aren't allowed to delete a referenced blob so it would only ever work right after a call to delete_all_jobs

This should be fine though because the RSC blob eviction algorithm ticks fairly frequently and all the abandoned blobs from a call to `delete_all_jobs` will be quickly reaped by it.